engine/src/main/java/org/destinationsol/assets/json/Validator.java

```diff
@@ -26,18 +27,24 @@
 
     static Logger logger = LoggerFactory.getLogger(Validator.class);
 
-    public static void validate(JSONObject json, String schemaPath) {
+    public static void validate(String jsonPath, String schemaPath) {
+        JSONObject json = Assets.getJson(jsonPath).getJsonValue();
         JSONObject schema;
 
         try {
             schema = Assets.getJson(schemaPath).getJsonValue();
         } catch (RuntimeException e) {
-            logger.warn("Json Schema " + schemaPath + " not found!", e);
+            logger.warn("Json Schema " + schemaPath + " not found!");
             return;
         }
 
         Schema schemaValidator = SchemaLoader.load(schema);
-        schemaValidator.validate(json);
+        try {
+            schemaValidator.validate(json);
+        }
+        catch(ValidationException e) {
```

Comment:
What is the point of this? If the json file is invalid then the game would have already crashed when trying to load the asset.
As an example, mess up the Hull Config json file of a ship and try to start the game with it.

Reply:
You did have a point, the game now throws a SolException when validation fails. Also the formatting should be fixed now.